cfme/tests/middleware/server_methods.py

```diff
@@ -92,12 +100,27 @@ def _get_server_by_name(provider, product=None, name=None):
     if len(servers) > 0:
         if name:
             for server in servers:
-                if re.match("^({})$".format(name), server.name):
+                if (re.match("^({})$".format(name), server.name) and
+                _check_container(server.hostname, container)):
                     return server
         else:
             return servers[0]
     raise ValueError('{} server was not found in servers list'.format(product))
 
 
+def _check_container(hostname, container):
+    """
+    Checks whether provided hostname is container or not when it is required or not.
+    """
```

Comment:
This function is rather confusingly written, perhaps change the name of the args or rewrite the docstring.